### PR TITLE
[MINOR] Make readme easier to follow

### PR DESCRIPTION
**Pull request by MrPowers**
### What changes were proposed in this pull request?

Update the README to make it easier to follow.

### Why are the changes needed?

I tried to get spark-connect-go running locally and it was a little confusing.  This new layout should make the setup steps a lot clearer.

### Does this PR introduce _any_ user-facing change?

Just updates the README.

### How was this patch tested?

N/A.

### CODE REVIEW

**Comment by nkarpov** [comment]
👍 nice I like it. I got tripped up bouncing between readme & quick start as a newbie.

File: README.md [hunk 1]
```diff
@@ -13,33 +12,42 @@ project reserves the right to withdraw and abandon the development of this proje
...
-> Ensure you have installed `buf CLI`; [more info](https://buf.build/docs/installation/)
 
-## How to write Spark Connect Go Application in your own project
+Step 4: Setup the Spark Driver on localhost.
 
-See [Quick Start Guide](quick-start.md)
+1. [Download Spark distribution](https://spark.apache.org/downloads.html) (3.4.0+), unzip the package.
 
-## Spark Connect Go Application Example
+2. Start the Spark Connect server with the following command:
```
Comment (grundprinzip) [inline README.md:35]:
```suggestion
2. Start the Spark Connect server with the following command (make sure to use a package version that matches your Spark distribution):
```

**Review by grundprinzip — APPROVED** [review]
Just a minor tweak.

**Review by grundprinzip — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (MrPowers) [reply README.md]: @grundprinzip - updated this language, good suggestion.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**